sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -542,6 +542,17 @@ object SQLConf {
       .longConf
       .createWithDefault(250 * 1024 * 1024)
 
+  val HIVE_FILESOURCE_FILE_CACHE_CONCURRENCY_LEVEL =
+    buildConf("spark.sql.hive.filesourceFileCacheConcurrencyLevel")
+      .doc("Provide concurrency level for underlying guava cache. The default is 1." +
+        " Higher value may give better performance, when the number of " +
+        "entries in the cache is very large. This imposes a limit on the size of single entry" +
+        "in the cache, i.e. sizeLimit < totalCacheCapacity/concurrencyLevel. See SPARK-27664.")
+      .intConf
+      .checkValue(value => value >= 1 && value <= 64, "The concurrency level should be " +
+        "between 1 and 64.")
+      .createWithDefault(1)
```

Comment:
Hi, @ScrapCodes . Could you propose this new feature to `master` branch first?

Reply:
Do we need to change the default value from `4` to `1`?